### ISSUE TITLE
Strip trailing whitespace

### ISSUE THREAD

**Issue by GoogleCodeExporter** [revelc/formatter-maven-plugin#23]
```
This doesn't appear to be an option in the eclipse formatter but is so useful 
it may be worth adding at this level. A number of tools really don't like 
trailing spaces (such as git) and it is an ambiguity in the formatting rules 
that can cause problems when patching if someones editor is too "helpful". I 
think this could be an "opt-in" option to start with.

It looks like it would be straightforward to add in at this line.

http://code.google.com/p/maven-java-formatter-plugin/source/browse/trunk/src/mai
n/java/com/relativitas/maven/plugins/formatter/FormatterMojo.java#453


if(stripWhitespace){
  formattedCode = formattedCode.replaceAll("\\s+$", "")
}
```

Original issue reported on code.google.com by `sam.hend...@gmail.com` on 18 Dec 2012 at 8:22

**Comment by hazendaz** [comment]
This is still present in Eclipse Mars as well.  What it appears is that Ecipse fixed the javadoc trailing space issue in source cleanup not formatting.  As such, since this project doesn't currently support source cleanup, we cannot easily fix this.

**Comment by ctubbsii** [comment]
Since this issue cannot be fixed without supporting executing other Eclipse tasks, we should close this for now. Maybe Eclipse will eventually support proper trailing whitespace stripping in the future.

**Comment by ctubbsii** [comment]
This was fixed by #521
